packages/material-ui/test/integration/Select.test.js

```diff
@@ -96,17 +94,31 @@ describe('<Select> integration', () => {
   });
 
   describe('with label', () => {
+    it('requires `id` and `labelId` for a proper accessible name', () => {
+      const { getByRole } = render(
+        <FormControl>
+          <InputLabel id="label">Age</InputLabel>
+          <Select id="input" labelId="label" value="10">
+            <MenuItem value="">none</MenuItem>
+            <MenuItem value="10">Ten</MenuItem>
+          </Select>
+        </FormControl>,
+      );
+
+      expect(getByRole('button')).to.have.accessibleName('Age Ten');
```

Comment:
Don't get distracted with the role (it's unclear whether this is correct). Nvda properly announces this as a combobox.

The important part is that we have a test for the accessible name of a select. I'll add a test to the `TextField` as well and to check how native vs non-native behaves (which requires verification with nvda).